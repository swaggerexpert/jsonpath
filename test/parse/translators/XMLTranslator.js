import { assert } from 'chai';
import { jestExpect as expect } from 'mocha-expect-snapshot';

import { parse, XMLTranslator } from '../../../src/index.js';

describe('parse', function () {
  context('translators', function () {
    context('XMLTranslator', function () {
      specify('should translate a JSONPath to a XML', function () {
        const parseResult = parse('$', { translator: new XMLTranslator() });

        assert.isTrue(parseResult.result.success);
        expect(parseResult.tree).toMatchSnapshot();
      });
    });
  });
});
