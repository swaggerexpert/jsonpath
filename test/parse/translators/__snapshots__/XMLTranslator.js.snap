// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse translators XMLTranslator should translate a JSONPath to a XML 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<root nodes="3" characters="1">
<!-- input string -->
  $
 <node name="jsonpath-query" index="0" length="1">
   $
  <node name="root-identifier" index="0" length="1">
    $
  </node><!-- name="root-identifier" -->
  <node name="segments" index="1" length="0">
    
  </node><!-- name="segments" -->
 </node><!-- name="jsonpath-query" -->
</root>
"
`;
